woocommerce/google-listings-and-ads
# Attribute Mapping - Add tracking

### Changes proposed in this Pull Request:

Closes part of #1648 .

This PR adds the tracking for the feature:

`gla_modal_closed`
`gla_modal_open`
`gla_attribute_mapping_save_rule_click`
`gla_attribute_mapping_delete_rule_click`
`gla_attribute_mapping_create_rule`
`gla_attribute_mapping_update_rule`
`gla_attribute_mapping_delete_rule`

### Detailed test instructions:


1. Enable event tracking logging by running `localStorage.setItem( 'debug', 'wc-admin:*' )` in the Console tab of DevTool, and refresh page to make it effective.
2. Go to Attribute Mapping page and click on Create rule button. It should trigger `gla_modal_open` with the context `attribute-mapping-create-rule-modal`.
3. Close the modal using cancel button, X button and clicking outside. It should trigger the event `gla_modal_closed` with the context `attribute-mapping-create-rule-modal` and the action `dismiss`
4.  Click again on Create rule button and click on save rule (after filling the form). It should trigger  `gla_attribute_mapping_save_rule_click` with context  `attribute-mapping-create-rule-modal` and also `gla_attribute_mapping_create_rule` with the same context and the rule props.
5. Click on Manage on the rule you just created. It should trigger  `gla_modal_open` with context  `attribute-mapping-manage-rule-modal`
6. Close the modal using cancel button, X button and clicking outside. It should trigger the event `gla_modal_closed` with the context `attribute-mapping-manage-rule-modal` and the action `dismiss`
7. Click again on Manage on the rule you just created and click on Save rule (after modifying the rule). It should trigger  `gla_attribute_mapping_save_rule_click` with context  `attribute-mapping-manage-rule-modal` and also `gla_attribute_mapping_update_rule` with the same context and the rule props.
8. Click on Delete on the rule you just created. It should trigger  `gla_modal_open` with context  `attribute-mapping-delete-rule-modal`
9. Close the modal using cancel button, X button and clicking outside. It should trigger the event `gla_modal_closed` with the context `attribute-mapping-delete-rule-modal` and the action `dismiss`
10. 8. Click again on Delete on the rule you just created. Confirm deletion. It should trigger  `gla_attribute_mapping_delete_rule_click` with context  `attribute-mapping-delete-rule-modal` and also `gla_attribute_mapping_delete_rule` with the same context and the rule props.


### Additional details:

Some  errors happened when I run `npm run doc:tracking`. Most of them are fixable by changing JSDOC definition but the one at the end:

```
google-listings-and-ads/node_modules/jsdoc-advanced-types-plugin/index.js:105
				throw "infinite loop!"
```

Requires more investigation and might be solved in future PRs
